api-server/scanner-lib/src/sync/tests.rs

```diff
@@ -1207,3 +1216,531 @@ async fn check_all_destinations_are_tracked(#[case] seed: Seed) {
+            assert!(!s_utxo.spent());
+            // check outputs are the same
+            assert_eq!(c_utxo.output(), s_utxo.output());
+        }
+        (None, Some(s_utxo)) => {
+            // if utxo is not found in chainstate but found in scanner it must be spent
+            assert!(s_utxo.spent());
+            // and not in locked utxos
+            assert_eq!(l_utxo, None);
+        }
+        (Some(c_utxo), None) => {
+            // if utxo is in chainstate but not in unlocked scanner utxos it must be in the locked
+            // ones
+            if let Some(l_utxo) = l_utxo {
+                assert_eq!(c_utxo.output(), &l_utxo.output);
+            } else {
+                panic!("Utxo in chainstate but not in the scanner state");
+            }
+        }
+        (None, None) => {
+            // on reorg utxos will be gone from both chainstate and the scanner
+            // same for locked
+            assert_eq!(l_utxo, None);
+        }
+    };
+}
+
+async fn check_staking_pools(
+    tf: &TestFramework,
+    local_state: &BlockchainState<TransactionalApiServerInMemoryStorage>,
+    staking_pools: &BTreeSet<PoolId>,
+) {
+    for pool_id in staking_pools {
+        check_pool(tf, local_state, *pool_id).await;
+    }
+}
+
+async fn check_pool(
+    tf: &TestFramework,
+    local_state: &BlockchainState<TransactionalApiServerInMemoryStorage>,
+    pool_id: PoolId,
+) {
+    let tx = local_state.storage().transaction_ro().await.unwrap();
+    let scanner_data = tx.get_pool_data(pool_id).await.unwrap().unwrap();
+
+    if let Some(node_data) = tf.chainstate.get_stake_pool_data(pool_id).unwrap() {
+        // check staker total balances are the same
+        assert_eq!(node_data.staker_balance(), scanner_data.staker_balance());
+        // check the pledges are the same
+        assert_eq!(node_data.pledge_amount(), scanner_data.pledge_amount());
+        assert_eq!(node_data, scanner_data);
+
+        // Compare the delegation shares
+        let node_delegations = tf
+            .chainstate
+            .get_stake_pool_delegations_shares(pool_id)
+            .unwrap()
+            .unwrap_or_default();
+
+        let scanner_delegations = tx.get_pool_delegations(pool_id).await.unwrap();
+
+        // check all delegations from the node are contained in the scanner
+        for (id, share) in &node_delegations {
+            let scanner_delegation = scanner_delegations.get(id).unwrap();
+            // check the shares are the same
+            assert_eq!(share, scanner_delegation.balance());
+        }
+
+        // delegations that have not been staked yet are stored in the scanner but not in the node
+        for (id, scanner_delegation) in scanner_delegations {
+            let share = node_delegations.get(&id).unwrap_or(&Amount::ZERO);
+            // check the shares are the same
+            assert_eq!(share, scanner_delegation.balance());
+        }
+    } else {
+        // the pool has been decommissioned
```

Comment:
- should the delegations be checked as well? they exist in the node still
- staker_balance balance is also 0 here